### PR TITLE
Normalize the api

### PR DESCRIPTION
**Pull request by vangheem**

### CODE REVIEW

File: guillotina_amqp/api.py [hunk 1]
```diff
@@ -1,11 +1,12 @@
...
+from guillotina.interfaces import IApplication
 from guillotina.response import HTTPNotFound
-from .state import TaskState
-from .state import get_state_manager
-from .exceptions import TaskNotFoundException
 
 
-@configure.service(method='GET', name='@amqp-tasks',
+@configure.service(method='GET', name='@amqp-tasks', context=IApplication,
                    permission='guillotina.ManageAMQP',
```
Comment (lferran) [inline guillotina_amqp/api.py]: shouldn't it be IContainer?

File: guillotina_amqp/job.py [hunk 2]
```diff
@@ -221,6 +221,7 @@ def state_manager(self):
         # caller
         #
 
+        aiotask_context.set('amqp_job', self)
```
Comment (lferran) [inline guillotina_amqp/job.py]: out of curiosity: why is this needed?

File: guillotina_amqp/api.py (hunk 1 above)
Comment (vangheem) [reply guillotina_amqp/api.py]: I think it's dangerous to do on IContainer.

These tasks are global across multiple containers. If you set it on IContainer, a container owner could self-assign local perms to have access to everyone else's tasks(Yes, I know often you can trust container owners).

File: guillotina_amqp/job.py (hunk 2 above)
Comment (vangheem) [reply guillotina_amqp/job.py]: I wanted a way for task to know about itself and this seemed like the most simple mechanism.

I'm open to other ideas!

File: guillotina_amqp/api.py (hunk 1 above)
Comment (vangheem) [reply guillotina_amqp/api.py]: If we want it maintained on IContainer, we'll have to iterate through them all and only provide the ones for this container.

Or, better yet maybe.. We prefix the task ids with the container id always--that sounds like a good idea...

Comment (lferran) [reply guillotina_amqp/api.py]: yes I agree! prefixing task ids with container ids its a must, in any case.

not sure about the implications of the endpoints at the IApplication or IContainer level though... aren't plugins installed on a container basis?

Comment (vangheem) [reply guillotina_amqp/api.py]: `applications` can affect the entire guillotina application.

`addons` are for only containers though. This guillotina application is not only for containers though!

Comment (vangheem) [reply guillotina_amqp/api.py]: Changed to filter on a prefix!
